### PR TITLE
Iteratively render tags

### LINKED ISSUE #6176: [2.10] - Fields containing {{type:Back}} no longer render
Reported on reddit

* Note Type: Basic

Front:
```
AA{{type:Back}}
```

* In 2.9, this renders as a "Type Answer" card.

* In 2.10, this shows `AA{{type:Back}}`

### PR DESCRIPTION
**Pull request by mikehardy**
## Purpose / Description
Tags in tags weren't rendering correctly, this showed up with a reported bug in expansion of tag that should have turned into a type-answer tag

## Fixes
Fixes #6176

## Approach
I applied the same transformation applied to section rendering to make it iterative until it determined no new replacements happened

## How Has This Been Tested?

With a test based on user report, used for the original bisect from @david-allison-1, integrated into TemplateTest

## Learning (optional, can help others)

Performance changes without a wide ranging test suite are fraught with regression danger.

## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/template/Template.java [hunk 1]
```diff
@@ -161,17 +154,34 @@ private String render_some_sections(@NonNull String template, @NonNull Map<Strin
...
-    private String render_tags(String template, Map<String, String> context) {
-        if (template.indexOf(ALT_HANDLEBAR_DIRECTIVE) != -1) {
+    private String render_some_tags(String template, Map<String, String> context) {
+        String ALT_HANDLEBAR_DIRECTIVE = "{{=<% %>=}}";
+        if (template.contains(ALT_HANDLEBAR_DIRECTIVE)) {
             template = template.replace(ALT_HANDLEBAR_DIRECTIVE, "").replace("<%", "{{").replace("%>", "}}");
         }
         StringBuffer sb = new StringBuffer();
         Matcher match = sTag_re.matcher(template);
+        boolean found = false;
```
Comment (mrozigor) [inline AnkiDroid/src/main/java/com/ichi2/libanki/template/Template.java:182]: I think that marking everything you can as final is a good practice.

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/template/Template.java:182]: Not if you're going to modify it. As we do with this one, it's not a constant.

Comment (mrozigor) [reply AnkiDroid/src/main/java/com/ichi2/libanki/template/Template.java:183]: Sorry, didn't wrote that it was general idea - as I can see a lot of code could be altered with `final` ;) But that's just some minor tweaks.

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/libanki/template/Template.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5527760&bid=18116912)

**Pull request merged**